api/libproject/environments.rb

```diff
@@ -0,0 +1,152 @@
+TEST_PROJECT = "all-of-us-workbench-test"
+
+def make_gae_vars(min_idle_instances = 0, max_instances = 10, instance_class = 'F1')
+  {
+    "GAE_MIN_IDLE_INSTANCES" => min_idle_instances.to_s,
+    "GAE_MAX_INSTANCES" => max_instances.to_s,
+    'GAE_INSTANCE_CLASS' => instance_class
+  }
+end
+
+def env_with_defaults(env, config)
+  {
+    :env_name => env,
+    :config_json => "config_#{env}.json",
+    :cdr_config_json => "cdr_config_#{env}.json",
+    :featured_workspaces_json => "featured_workspaces_#{env}.json",
+    :gae_vars => make_gae_vars,
+  }.merge(config)
+end
+
+# TODO: Make environment/project flags consistent across commands, consider
+# using environment keywords as dict keys here, e.g. :test, :staging, etc.
+ENVIRONMENTS = {
+  "local" => env_with_defaults("local", {
+    :api_endpoint_host => "localhost:8081",
+    :cdr_sql_instance => "workbench",
+    :source_cdr_project => "all-of-us-ehr-dev"
+  }),
+  "all-of-us-workbench-test" => env_with_defaults("test", {
+    :api_endpoint_host => "api-dot-#{TEST_PROJECT}.appspot.com",
+    :cdr_sql_instance => "#{TEST_PROJECT}:us-central1:workbenchmaindb",
+    :gae_vars => make_gae_vars(0, 10, 'F4'),
+    :publisher_account => "circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com",
+    :source_cdr_project => "all-of-us-ehr-dev",
+    :source_cdr_wgs_project => "all-of-us-workbench-test",
+    :accessTiers => {
+      "registered" => {
+        :ingest_cdr_project => "fc-aou-vpc-ingest-test",
+        :dest_cdr_project => "fc-aou-cdr-synth-test",
+        :auth_domain_group_email => "GROUP_all-of-us-registered-test@dev.test.firecloud.org",
+      },
+      "controlled" => {
+        :ingest_cdr_project => "fc-aou-cdr-ingest-test-2",
+        :dest_cdr_project => "fc-aou-cdr-synth-test-2",
+        :auth_domain_group_email => "all-of-us-test-prototype-3@dev.test.firecloud.org",
+      }
+    }
+  }),
+  "all-of-us-rw-staging" => env_with_defaults("staging", {
+    :api_endpoint_host => "api-dot-all-of-us-rw-staging.appspot.com",
+    :cdr_sql_instance => "#{TEST_PROJECT}:us-central1:workbenchmaindb",
+    :gae_vars => make_gae_vars(0, 10, 'F2'),
+    :source_cdr_project => "all-of-us-ehr-dev",
+    :publisher_account => "circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com",
```

Comment:
I know you just copied this over but is this the right publisher account? using test and not staging?

Reply:
Yes, it's correct for test, staging, perf - these envs all get pushed via CircleCI so that is the primary deploy account. Technically the publish-cdr flow doesn't happen from there, so could use a different SA, but using this probably makes sense for consistency